### PR TITLE
Fixes an issue where it was not possible to invite an existing user whose userid was not equal to his loginname.

### LINKED ISSUE #50: Inviting users with different login-name and id will fail.
Inviting a user in the @@invite-view will fail as soon as the users id differs to its login-name.

bildschirmfoto 2017-04-04 um 17 35 14

https://github.com/4teamwork/ftw.participation/blob/master/ftw/participation/browser/invite.py#L241

```
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module plone.z3cform.layout, line 66, in __call__
  Module plone.z3cform.layout, line 50, in update
  Module ftw.participation.browser.invite, line 194, in update
  Module z3c.form.form, line 210, in update
  Module plone.app.z3cform.csrf, line 21, in execute
  Module z3c.form.action, line 98, in execute
  Module z3c.form.button, line 315, in __call__
  Module z3c.form.button, line 170, in __call__
  Module ftw.participation.browser.invite, line 241, in handle_invite
AttributeError: 'NoneType' object has no attribute 'getProperty'
```

Issuer: https://extranet.4teamwork.ch/support/agg/tracker/134

### PR DESCRIPTION
**Pull request by elioschmutz**
This PR fixes an issue where it was not possible to invite an existing user whose userid was not equal to his login-name.

Closes #50

### CODE REVIEW

File: ftw/participation/tests/test_invite.py [hunk 1]
```diff
@@ -111,6 +112,28 @@ def test_inviting_a_user_by_userid(self, browser):
         self.assertIn('Hi th=C3=B6re', mail)
 
     @browsing
+    def test_inviting_a_user_by_userid_with_a_different_login_name_is_possible(self, browser):
+        user = create(Builder('user')
+                      .named('Hugo', 'Boss')
+                      .with_userid('hugo.boss')
+                      .with_email('hugo.boss@example.ch'))
+        api.portal.get_tool('acl_users').source_users.updateUser('hugo.boss', 'hugo.boss@example.ch')
```
Comment (mbaechtold) [inline ftw/participation/tests/test_invite.py]: As I don't know why this is necessary (lack of knowledge about `acl_users`), would it be useful to explain the reason for this in a code comment?

Comment (elioschmutz) [inline ftw/participation/tests/test_invite.py]: 👍  I added a comment who is explaining how we do achieve this issue in a real plone-site.

Comment (jone) [inline ftw/participation/tests/test_invite.py]: @elioschmutz but why do you call `source_users.updateUser`?
The builder above already sets the userid and the email.

Comment (elioschmutz) [inline ftw/participation/tests/test_invite.py]: it's not the `userid` and the `email`, it's the `userid` and the `login_name`.

See https://github.com/plone/Products.PlonePAS/blob/master/src/Products/PlonePAS/plugins/property.py#L272

**Pull request merged**